### PR TITLE
Adding Appsync realtime provider on PubSub

### PR DESCRIPTION
**Pull request by elorzafe**
_Issue #, if available:_

_Description of changes:_
Adding Appsync realtime provider on PubSub
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/api/__tests__/API-test.ts [hunk 1]
```diff
@@ -560,7 +564,7 @@ describe('API test', () => {
 				})
 			).rejects.toThrowError('No federated jwt');
 		});
-		test('multi-auth using CUP as auth mode, but no userpool', async () => {
+		test.skip('multi-auth using CUP as auth mode, but no userpool', async () => {
```
Comment (sammartinez) [inline packages/api/__tests__/API-test.ts]: Why are we skipping the test here?

File: packages/core/__tests__/Util-test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+				}
+			};
+		}
+
+		const retryableFunction = createRetryableFunction();
+
+		try {
+			jitteredExponentialRetry(retryableFunction, []).then(done);
+		} catch (err) {}
+		// jest.runAllTimers();
```
Comment (sammartinez) [inline packages/core/__tests__/Util-test.ts]: Is this needed?

File: packages/core/__tests__/Util-test.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+import { jitteredExponentialRetry, NonRetryableError } from '../src/Util';
+import { ConsoleLogger as Logger } from '../src/Logger';
+Logger.LOG_LEVEL = 'DEBUG';
+describe('Util', () => {
+	beforeEach(() => {
+		// jest.useFakeTimers();
+		// jest.clearAllMocks();
```
Comment (sammartinez) [inline packages/core/__tests__/Util-test.ts]: Why are commenting these out?

File: packages/core/__tests__/Util-test.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		const retryableFunction = createRetryableFunction();
+
+		try {
+			jitteredExponentialRetry(retryableFunction, []).then(done);
+		} catch (err) {}
+		// jest.runAllTimers();
+	});
+	test('Fail with NonRetryableError', async () => {
+		const isNonRetryableError = (obj: any): obj is NonRetryableError =>
+			obj instanceof NonRetryableError;
```
Comment (Amplifiyer) [inline packages/core/__tests__/Util-test.ts]: Do we need this?

File: packages/core/__tests__/Util-test.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+		console.log(nonRetryableError.constructor.name);
+		const testFunc = jest.fn(() => {
+			throw nonRetryableError;
+		});
+		expect.assertions(1);
+
+		try {
+			await jitteredExponentialRetry(testFunc, []);
+		} catch (err) {
+			// expect(err).toBe(nonRetryableError);
```
Comment (Amplifiyer) [inline packages/core/__tests__/Util-test.ts]: I think we should test/assert this

File: packages/core/src/Util.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
+import { DelayFunction } from './types';
+import { ConsoleLogger as Logger } from './Logger/ConsoleLogger';
+const logger = new Logger('Util');
+
+export class NonRetryableError extends Error {
+	public readonly nonRetryable = true;
+	constructor(message: string) {
+		super(message);
+	}
+}
```
[the remark this answers is not in the document]
Comment (elorzafe) [reply packages/core/src/Util.ts]: this is not only relevant for TS but for JS as well

File: packages/core/__tests__/Util-test.ts (hunk 5 above)
Comment (elorzafe) [reply packages/core/__tests__/Util-test.ts]: done

File: packages/core/__tests__/Util-test.ts (hunk 4 above)
Comment (elorzafe) [reply packages/core/__tests__/Util-test.ts]: removed

File: packages/core/__tests__/Util-test.ts (hunk 2 above)
Comment (elorzafe) [reply packages/core/__tests__/Util-test.ts]: removed

**Pull request merged**